herbert_core/utilities/hdf_nexus/find_root_nexus_dir.m

```diff
@@ -61,26 +72,28 @@
     return;
 end
 if(n_nx_entries>1)
-    error('ISIS_UTILITES:invalid_argument',' found multiple nxspe folders in file %s but this function does not currently support multiple nxspe folders',hdf_fileName)
+    if ~test_mode
+        error('ISIS_UTILITES:invalid_argument',...
+            ' found multiple nxspe folders in file %s but this function does not currently support multiple nxspe folders',hdf_fileName)
+    end
+    root_nx_path = nx_folders;
+    data_version    = nx_version;
+else
+    root_nx_path = nx_folders{1};
+    data_version    = nx_version{1};
 end
-root_nx_path = nx_folders{1};
-data_version    = nx_version{1};
-
 
-function ver=read_nxspe_version(hdf_fileName,DS)
+function ver=get_version(def_dataset,filename)
 % Matlab version specific function to obtain correct NeXus version
-mat_ver_array=datevec(version('-date'));
-
-if mat_ver_array(1)<=2009
-    [fp,shortName] = fileparts(DS.Attributes.Name);                    
-    ver = hdf5read(hdf_fileName,[DS.Name,'/',shortName]);
-
-else
-   ver = hdf5read(hdf_fileName,DS.Name, DS.Attributes.Shortname);
+%mat_ver_array=datevec(version('-date'));
```

Comment:
is this code still required?

Reply:
there is thought on checking it agains old matlab, but probably not worth it. Remved.